pulp_ansible/app/tasks/collections.py

```diff
@@ -608,13 +661,60 @@ async def _find_all_collections(self):
 
         await asyncio.gather(*tasks)
 
+    async def _should_we_sync(self):
+        """Check last synced metadata time."""
+        root, api_version = await self._get_root_api(self.remote.url)
+        if api_version == 3:
+            downloader = self.remote.get_downloader(
+                url=root, silence_errors_for_response_status_codes={404}
+            )
+            try:
+                metadata = parse_metadata(await downloader.run())
+            except FileNotFoundError:
+                return True
+
+            try:
+                self.last_synced_metadata_time = parse_datetime(metadata["published"])
+            except KeyError:
+                return True
+
+            sources = set()
+            if self.collection_info:
+                sources = {r.source for r in self.collection_info if r.source}
+            sources.add(self.remote.url)
+            if len(sources) > 1:
+                return True
+
+            if self.last_synced_metadata_time == self.repository.last_synced_metadata_time:
+                return False
+
+        return True
+
     async def run(self):
         """
         Build and emit `DeclarativeContent` from the ansible metadata.
         """
+        if self.repository.remote and self.remote.pk == self.repository.remote.pk:
+            should_we_sync = await self._should_we_sync()
+            msg = _("no-op: Checking if remote changed since last sync.")
+            noop = ProgressReport(message=msg, code="noop")
+            noop.state = TASK_STATES.COMPLETED
+            if self.optimize and not should_we_sync:
+                noop.message = _(
+                    "no-op: {remote} did not change since last sync - {published}".format(
+                        remote=self.remote.url, published=self.last_synced_metadata_time
+                    )
+                )
+                noop.save()
+                log.debug(_("no-op: remote wasn't updated since last sync."))
+                return
+            noop.save()
```

Comment:
What do you think about moving this into the `_should_we_sync` and then have this be:

```
if self.optimize:
    should_we_sync = await self._should_we_sync():
    if should_we_sync is False:
        return
    
```

The reasoning is to keep the concerns of the no-op in the area where the no-op happens.